zcash_client_sqlite/src/wallet.rs

```diff
@@ -592,13 +618,13 @@ pub(crate) fn truncate_to_height<P: consensus::Parameters>(
     // nothing to do if we're deleting back down to the max height
     if block_height < last_scanned_height {
         // Decrement witnesses.
-        wdb.conn.execute(
+        conn.execute(
             "DELETE FROM sapling_witnesses WHERE block > ?",
             [u32::from(block_height)],
         )?;
 
         // Rewind received notes
-        wdb.conn.execute(
+        conn.execute(
             "DELETE FROM sapling_received_notes
                 WHERE id_note IN (
```

Comment:
In some SQL implementations you can [directly delete from an outer join](https://stackoverflow.com/a/5808865/393146) (specifying which table is to be deleted from). I don't know whether this is supported by SQLite.